Don't use -latomic on macOS. Clang on macOS doesn't use libatomic.

https://github.com/nodejs/node/issues/30093

This bug caused the build to fail when using MacPorts clang. Due to a second
bug, in which node's build system misidentifies clang from Xcode 7 and later
as "not clang", it didn't add -latomic for Xcode clang and thus the build
succeeded.
--- node.gyp.orig	2019-10-23 15:09:52.000000000 -0500
+++ node.gyp	2019-10-23 15:27:25.000000000 -0500
@@ -290,7 +290,7 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['OS in ("linux", "mac") and llvm_version != "0.0"', {
+      ['OS in ("linux") and llvm_version != "0.0"', {
         'libraries': ['-latomic'],
       }],
     ],
